pkg/controller/jobs/job/job_controller_test.go

```diff
@@ -450,6 +450,66 @@ func TestReconciler(t *testing.T) {
 					Obj(),
 			},
 		},
+		"when workload is not admitted due to Active field being false and job is suspended": {
+			job: *baseJobWrapper.Clone().
+				Obj(),
+			wantJob: *baseJobWrapper.Clone().
+				Suspend(true).
+				Obj(),
+			workloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("a", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 10).Request(corev1.ResourceCPU, "1").Obj()).
+					ReserveQuota(utiltesting.MakeAdmission("cq").AssignmentPodCount(10).Obj()).
+					Admitted(true).
+					Active(false).
+					AdmissionCheck(kueue.AdmissionCheckState{
+						Name:  "check",
+						State: kueue.CheckStateReady,
+						PodSetUpdates: []kueue.PodSetUpdate{
+							{
+								Name: "main",
+								Labels: map[string]string{
+									"ac-key": "ac-value",
+								},
+							},
+						},
+					}).
+					Obj(),
+			},
+			wantWorkloads: []kueue.Workload{
```

Comment:
This is not ideal. We generally don't want to directly set the admitted condition to False. Instead, we add an Eviction condition and set suspend to false.

A future reconcile would observe that the Job actually paused, and only then we would swap the Admitted condition to false.

So overall you can split this into 2 test cases, where the second test case is the second reconcile described above.